plugins/woocommerce/src/Internal/Utilities/BatchProcessor.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Implements abstract class for action scheduler based batch processing. Updates using this class must be tolerant of same record getting processed more than once.
+ */
+
+namespace Automattic\WooCommerce\Internal\Utilities;
+
+/**
+ * Class BatchProcessor
+ *
+ * @package Automattic\WooCommerce\DataBase
+ */
+abstract class BatchProcessor {
+
+	/**
+	 * A user friendly name for this process.
+	 *
+	 * @return string Name of the process.
+	 */
+	abstract protected function get_name() : string;
+
+	/**
+	 * A user friendly description for this process.
+	 *
+	 * @return string Description.
+	 */
+	abstract protected function get_description() : string;
+
+	/**
+	 * Process data for current batch.
+	 *
+	 * @param array $batch Batch details.
+	 */
+	abstract public function process_for_batch( array $batch );
```

Comment:
This method name may be confusing: "process for batch" can be interpreted as "give me the process needed for the batch" instead of "go an actually process the batch". So I think `process_batch` would be better.

Reply:
👍  renamed in https://github.com/woocommerce/woocommerce/pull/33233/files#diff-1e070cc170d3b65cd6ba61b60cbd0697c8431fda674b4720926c8fbc1afecce7R297